decode/src/lib.rs

```diff
@@ -669,13 +669,84 @@ impl ParseFromModule for MemoryModel {
     }
 }
 
+/// Description of software power control override status for a module.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct PowerControl {
+    /// If true, the module is configured for software control of low power
+    /// mode. In this case, the `LPMode` pin has no effect.
+    pub software_override: bool,
+    /// If true, the module is set to low power mode by software. If `false`,
+    /// then the module is allowed to enter high power mode.
+    ///
+    /// This is _only meaningful_ if `software_override` is `true`.
+    pub low_power: bool,
+}
```

Comment:
If `low_power` is only meaningful when `software_override` is true, that suggests this could be

```suggestion
pub enum PowerControl {
    /// The module is configured to use the pin to select low power mode
    UseLpModePin,

    /// The module is configured for software control of low power mode
    OverrideLpModePin {
        /// If true, the module is set to low power mode by software. If
        /// `false`, then the module is allowed to enter high power mode.
        low_power: bool,
    },
}
```

Dunno if it's worth making the resulting changes, but this seems like a more faithful model of what you're trying to encode.

Reply:
That does seem better, thanks!